### ISSUE TITLE
[Flaky Test] Controller Common Suite

### ISSUE THREAD

**Issue by timebertt** [gardener/gardener#4708]
**How to categorize this issue?**

/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:

`Controller Common Suite` (`github.com/gardener/gardener/extensions/pkg/controller/common`).

**CI link**:

https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/58#L613d21f7:271
https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-head-update-job/builds/61#L613d25f9:272
https://concourse.ci.gardener.cloud/teams/gardener/pipelines/gardener-master/jobs/master-pull-request-job/builds/150#L613d54d5:300

**Reason for failure**:

Seems to be a timeout, the test is time-sensitive:

```
Running Suite: Controller Common Suite
======================================
Random Seed: 1632294277
Will run 18 of 18 specs

•••••••panic: test timed out after 2m0s

goroutine 28 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1700 +0x11f
created by time.goFunc
	/usr/local/go/src/time/sleep.go:180 +0x52

goroutine 1 [chan receive, 2 minutes]:
testing.(*T).Run(0xc00049b080, 0x2c35558, 0xa, 0x2d06038, 0x0)
	/usr/local/go/src/testing/testing.go:1239 +0x610
testing.runTests.func1(0xc00049b080)
	/usr/local/go/src/testing/testing.go:1511 +0xa7
testing.tRunner(0xc00049b080, 0xc000335ce0)
	/usr/local/go/src/testing/testing.go:1193 +0x203
testing.runTests(0xc00000f1b8, 0x3e21110, 0x1, 0x1, 0xc04ad35f652f519a, 0x1c0885f5cf, 0x3e60200, 0xc0001a2148)
	/usr/local/go/src/testing/testing.go:1509 +0x613
testing.(*M).Run(0xc0003ca400, 0x0)
	/usr/local/go/src/testing/testing.go:1417 +0x3b4
main.main()
	_testmain.go:45 +0x237

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x3e606a0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:1169 +0xb1
created by k8s.io/klog/v2.init.0
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/v2/klog.go:420 +0x1a7

goroutine 5 [chan receive]:
k8s.io/klog.(*loggingT).flushDaemon(0x3e605c0)
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/klog.go:1010 +0xb1
created by k8s.io/klog.init.0
	/go/src/github.com/gardener/gardener/vendor/k8s.io/klog/klog.go:411 +0x190

goroutine 85 [select]:
github.com/gardener/gardener/extensions/pkg/controller/common.(*checkerWatchdog).Start.func1(0x2f5a6f8, 0xc0005aeb00, 0xc00030a0c0, 0x2f54db8, 0xc0005aeb40)
	/go/src/github.com/gardener/gardener/extensions/pkg/controller/common/checkerwatchdog.go:89 +0x1de
created by github.com/gardener/gardener/extensions/pkg/controller/common.(*checkerWatchdog).Start
	/go/src/github.com/gardener/gardener/extensions/pkg/controller/common/checkerwatchdog.go:86 +0x1db

goroutine 83 [chan receive]:
github.com/gardener/gardener/extensions/pkg/controller/common.(*checkerWatchdog).Result(0xc00030a0c0, 0x2f48c00, 0x0, 0x0)
	/go/src/github.com/gardener/gardener/extensions/pkg/controller/common/checkerwatchdog.go:183 +0xbe
github.com/gardener/gardener/extensions/pkg/controller/common_test.glob..func1.3.2()
	/go/src/github.com/gardener/gardener/extensions/pkg/controller/common/checkerwatchdog_test.go:91 +0x657
github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc0003fa420, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/leafnodes/runner.go:113 +0xfd
github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc0003fa420, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/leafnodes/runner.go:64 +0x185
github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0xc000696440, 0x2f12dc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/leafnodes/it_node.go:26 +0xba
github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc000320870, 0x0, 0x2f12dc0, 0xc0001a5080)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/spec/spec.go:215 +0x7fe
github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc000320870, 0x2f12dc0, 0xc0001a5080)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/spec/spec.go:138 +0x188
github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc0004a4000, 0xc000320870, 0x0)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:200 +0x17c
github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc0004a4000, 0x1)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:170 +0x236
github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc0004a4000, 0x8)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:66 +0x146
github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc0001e64d0, 0x7f5f12597de0, 0xc00049b200, 0x2c487af, 0x17, 0xc0005bf8f0, 0x1, 0x1, 0x2f618b8, 0xc0001a5080, ...)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:79 +0x83a
github.com/onsi/ginkgo.runSpecsWithCustomReporters(0x2f13da0, 0xc00049b200, 0x2c487af, 0x17, 0xc000090e70, 0x1, 0x1, 0x7f5f3c11b5b8)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:238 +0x358
github.com/onsi/ginkgo.RunSpecs(0x2f13da0, 0xc00049b200, 0x2c487af, 0x17, 0x0)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:213 +0xe5
github.com/gardener/gardener/extensions/pkg/controller/common_test.TestCommon(0xc00049b200)
	/go/src/github.com/gardener/gardener/extensions/pkg/controller/common/common_suite_test.go:26 +0x109
testing.tRunner(0xc00049b200, 0x2d06038)
	/usr/local/go/src/testing/testing.go:1193 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1238 +0x5d8

goroutine 59 [chan receive, 1 minutes]:
github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).registerForInterrupts(0xc0004a4000, 0xc00004ecc0)
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:223 +0xf4
created by github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run
	/go/src/github.com/gardener/gardener/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:60 +0xaf

goroutine 61 [syscall, 1 minutes]:
os/signal.signal_recv(0x4a8801)
	/usr/local/go/src/runtime/sigqueue.go:168 +0xa5
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x30
created by os/signal.Notify.func1.1
	/usr/local/go/src/os/signal/signal.go:151 +0x7d
FAIL	github.com/gardener/gardener/extensions/pkg/controller/common	120.480s
```

**Comment by timebertt** [comment]
/cc @stoyanr 
The test was introduced in #4638.

**Comment by stoyanr** [comment]
Fixed with #4752
